src/com/esotericsoftware/kryo/unsafe/UnsafeUtil.java

```diff
@@ -124,7 +125,7 @@ public static ByteBuffer newDirectBuffer (long address, int size) {
 		if (directByteBufferConstructor == null)
 			throw new UnsupportedOperationException("No direct ByteBuffer constructor is available.");
 		try {
-			return directByteBufferConstructor.newInstance(address, size, null);
+			return directByteBufferConstructor.newInstance(address, size);
 		} catch (Exception ex) {
 			throw new KryoException("Error creating a ByteBuffer at address: " + address, ex);
```

Comment:
This change is unrelated to records, but is clearly required to successfully run on JDK 14+, since the non-Public constructors of DirectByteBuffer has been refactored / changed.